interview-questions.md

```diff
@@ -8,48 +8,59 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is JSX? What is one syntax difference between HTML and JSX?
 
-Your answer:
+Your answer: JSX is a coding language that is a hybrid of HTML and Javascript. JSX makes it easier to write react application by allowing you to simply access different elements 
 
-Researched answer:
+Researched answer: After some research, I understand that JSX is indeed a way to write HTML and Javascript together. JSX allows for object oriented programming (OOP relies on the concept of classes and objects - used to create simple, reusable pieces of code blueprints, aka classes, to create individual instances of objects). The biggest syntactical different between HTML and JSX is that JSX requires the use of parent/child components. The code won't compile without this, whereas HTML does not require you to return everything to s single parent element. 
 
 2. What is yarn? What file(s) are modified in a React application when you run yarn?
 
-Your answer:
+Your answer: Yarn is Yet Another Resource Navigator
 
-Researched answer:
+Researched answer: I was so close! Yet Another Resource Negotiator is the official name, and it was created in 2016 in a collaboration between Meta, Google and a few other companies. Yarn is a package manager, which means that it is a collection of software tools that automate different processes (installing, upgrading, configuring and removing) for computer programs consistently. This increases stability, and it is also designed to eliminate the need for manual installs and updates. Yarn creates a json file and node modules.
 
 3. What is an anonymous function in JavaScript?
 
-Your answer:
+Your answer: () => {}
 
-Researched answer:
+Researched answer: An anonymous function is aptly names because it is a function that has no name. It can only be accessed by the variable stored in the function as the value, and cannot be accessed after it's creation.
 
 4. What is the difference between a class and an object?
 
-Your answer:
+Your answer: A class is used to create objects, and an object can be a part of a class.
 
-Researched answer:
+Researched answer: Classes are templates for objects, and require the use of the constructor method. According to freeCodeCamp, classes in JS are an example of syntactical sugar vs the prototype based inheritance model used to implement object oriented programming.
```

Comment:
Good stuff! Prototype based inheritance model is really weird and interesting if you want to get into it.